### ISSUE TITLE
Docket Clerk: Select party/parties to be served

### ISSUE THREAD

**Issue by JessicaMarine** [flexion/ef-cms#7119]
eFiled
Paper Filed
Court-issued

**Comment by klohman** [comment]
UX Research Questions 
* Tell us about how your process (current and legacy) to serve individual parties. 
* How often do you need to serve this happen? 
* What determines who should be served?
* Is there added risk in only serving individual parties? 
* What happens if you make a mistake? Do you ever need to go back and serve another party?

**Comment by cholly75** [comment]
@klohman - Use cases for choosing "none": return mail and trial minutes are both docket additions that should not be served when filed.

**Comment by klohman** [comment]
## Add Paper Filing - No UI Change

**On Save and Serve**
All parties other than the filing party are served
Filing party or parties are not served  
Docket entry reflects served date and the parties served 
P - if only petitioner, intervenor, participant, or counsel was served 
R - if respondent only was served 
B -if both any petitioner, ntervenor, participant, or counsel AND respondent were served 

**Jointly filed documents**
if all parties file a document, no service is initiated 
no served stamp is applied to the document 
docket record does not display a served date or served party and does not display “Not served” message

## Add Docket Entry (Court-Issued Document) - No Change

**Service Parties**
* no change 
* all parties are served for all court-issued documents

## Add Docket Entry (Court-Issued Document - isUnservable)

For any court-issued event code with isUnservable 

![image.png](https://images.zenhubusercontent.com/5bd7271508843955cc24d3d0/2b6fca78-3a8f-4371-ab79-aca86ef0a8f7)

**Service Parties**
Display message “This document type does not require service”

**On Save**
No change to current functionality

**Comment by klohman** [comment]
## Tests

#### 1) Docket clerk adds paper-filing and non-filing parties have all electronic service 
* Login as docket clerk 
* Go to case 
    * Case has at least 2 petitioners 
        * petitioner1
        * petitioner2
* All parties on case have electronic service 
* Click Add Paper Filing 
* Select filing party = petitioner1, petitioner2
* Save and Serve
Expected results 
- [ ] Document is added to the docket record with Served date = [current date] and Parties = R
- [ ] Respondent receives service email 
- [ ] petitioner1 does not receive service email 
- [ ] petitioner2 does not receive service email 

#### 2) Docket clerk adds paper-filing and non-filing parties have all paper service  
* Login as docket clerk 
* Go to case 
    * Case has at least 2 petitioners 
        * petitioner1 - paper service
        * petitioner2 - paper service 
    * Case has at least 1 respondent counsel - electronic service
* Click Add Paper Filing 
* Select filing party = respondent
* Save and Serve
Expected results 
- [ ] Document is added to the docket record with Served date = [current date] and Parties = P
- [ ] Respondent does not receive service email 
- [ ] App routes to Print Paper Service screen with document PDF 
- [ ] PDF contains address sheets for petitioner1 and petitioner2 

#### 3) Docket clerk adds paper-filing and non-filing parties have both electronic and paper service   
* Login as docket clerk 
* Go to case 
    * Case has at least 2 petitioners 
        * petitioner1 - paper service
        * petitioner2 - electronic service 
    * Case has at least 1 respondent counsel - electronic service
* Click Add Paper Filing 
* Select filing party = respondent
* Save and Serve
Expected results 
- [ ] Document is added to the docket record with Served date = [current date] and Parties = P
- [ ] Respondent does not receive service email 
- [ ] Petitioner2 receives service email 
- [ ] App routes to Print Paper Service screen with document PDF 
- [ ] PDF contains address sheets for petitioner1

#### 4) Docket clerk adds paper-filing and non-filing parties include intervenor/participants 
* Login as docket clerk 
* Go to case 
    * Case has petitioner1 - electronic service
    * Case has intervenor - electronic service 
    * Case has participant - paper service 
* Click Add Paper Filing 
* Select filing party = petitioner1, respondent 
* Save and Serve 
Expected results 
- [ ] Document is added to the docket record with Served date = [current date] and Parties = P
- [ ] Respondent does not receive service email 
- [ ] Petitioner1 does not receive service email 
- [ ] Intervenor receives service email   
- [ ] App routes to Print Paper Service screen with document PDF 
- [ ] PDF contains address sheets for participant

#### 5) Docket clerk adds paper-filing and non-filing party is respondent with multiple respondent counsel
* Login as docket clerk 
* Go to case 
    * Case has multiple respondent counsel
        * respondent1- electronic service 
        * respondent2 - electronic service 
* Click Add Paper Filing 
* Select Filing party = petitioner 
* Save and Serve 
Expected results 
- [ ] Document is added to the docket record with Served date = [current date] and Parties = R
- [ ] Petitioner does not receive service 
- [ ] respondent1 receives service email 
- [ ] respondent2 receives service email 

#### 6) Docket clerk adds paper-filing jointly  filed by all parties 
* Go to case 
    * Case has petitioner1 - electronic service
    * Case has intervenor - electronic service 
    * Case has participant - paper service 
    * Case has respondent counsel
* Click Add Paper Filing 
* Select filing party = all parties 
* Save and Serve 
Expected results 
- [ ] Document is added to the docket record with Served date = [blank] and Parties = [blank] 
- [ ] Docket entry does not display “Not Served” message
- [ ] Coversheet does not display a Served stamp
- [ ] Respondent does not receive service email 
- [ ] Petitioner1 does not receive service email 
- [ ] Intervenor does not receive service email   
- [ ] App does not routes to Print Paper Service screen with document PDF 

#### 7) Docket clerk adds and serves court-issued document 
* Login as docket clerk 
* Go to case 
    * Case has multiple petitioners
    * Case has multiple respondent counsel
        * respondent1- electronic service 
        * respondent2 - electronic service 
* Create, sign and add docket entry for Order 
* Save and Serve 
Expected results 
- [ ] All parties receive service 
- [ ] Document is added to the docket record with Served date = [current date] and Parties = B

#### 8) Docket clerk adds court-issued document that is unservable 
* Login as docket clerk 
* Go to case 
* Upload PDF, Add Docket Entry 
* Select document type = U.S.C.A.
* Save Entry 
Expected results 
On select of document type USCA
- [ ] Service parties field displays message “This document does not require service.”
On Save 
- [ ] Document is added to the docket record with Served date and Parties columns blank

**Comment by matthopson** [comment]
@klohman I think I know the answer to this, but just want to be clear.... In a scenario like where a Petitioner is filing a document, they would not be served - but I _assume_ the intent is to show a P (even though they weren't served, they were the only party that _would_ be served if they weren't the filing party). 🥴

**Comment by JessicaMarine** [comment]
@matthopson No.  If P files a document, then they wouldn't be served.  So, the served party would be R (or blank if nobody is served).

**Comment by matthopson** [comment]
@JessicaMarine  Thanks!

**Comment by mariahkannenberg** [comment]
Was in progress. Diverted to a bug. Move to batch 10. Matt can revert the change. 
Anyone who picks up this story can talk to Matt about what he found. This is a "sticky" story - and has some assumptions.
